### PR TITLE
feat(doc) Add doc generation with sphinx

### LINKED ISSUE #7: [doc] Add documentation generation with sphinx
Add documentation with sphynx:

- [ ] create a root directory `documentation`
- [ ] add `conf.py` to define sphynx option 
- [ ] TO BE DISCUSSED : move all .md documentation files to the  `documentation directory 
- [ ] create landing page `documentation/index.md`

### PR DESCRIPTION
**Pull request by benoitblanc**
closes #7 

To build doc:

```bash
# install aditionnal dependencies
python -m pip install -U -r requirements/documentation.txt
# build it
sphinx-build -b html docs docs/_build
```

### CODE REVIEW

File: docs/development/documentation.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+Sphinx est utilisé pour la génération de la documentation depuis des pages écrites en Markdown (via le [parser MyST](https://myst-parser.readthedocs.io/en/latest/)).
+
+## Génération du site web de documentation
+
+Pour la génération:
+
+```bash
+# install aditionnal dependencies
+python -m pip install -U -r requirements/documentation.txt
```
Comment (jmkerloch) [inline docs/development/documentation.md:11]: We also need to install base requirements.txt.

Can you move the requirements.txt file to the requirements folder and add it in the documentation for documentation build ?

The base requirements.txt also need to be updated to use the same packages as the ones described in docker/debian/Dockerfile. (error with psycopg2, we need to use psycopg2-binary

File: docs/docker/readme.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Utiliser r2gg avec Docker 
+
+Le dossier {{ '[docker]({}tree/{}/docker)'.format(repo_url, repo_branch) }} regroupe les différents fichiers permettant d'utiliser r2gg avec docker. 
+
+Les sous-dossiers sont ordonnés ainsi pour des raisons pratiques : 
+- {{ '[scripts]({}tree/{}/docker/scripts)'.format(repo_url, repo_branch) }} : Ce dossier regroupe des scripts utiles dans les différents docker. 
+- {{ '[config]({}tree/{}/docker/config)'.format(repo_url, repo_branch) }} : Ce dossier regroupe plusieurs fichiers de configurations qui se trouvent être communs aux autres sous-dossiers. 
+- {{ '[debian]({}tree/{}/docker/debian)'.format(repo_url, repo_branch) }} : Pour utiliser r2gg sur debian 10
```
Comment (jmkerloch) [inline docs/docker/readme.md:8]: No need to use docker/debian/readme in repository. You can use the docs/docker/debian/readme.md file

**Review by jmkerloch — CHANGES_REQUESTED** [review]

File: docs/docker/readme.md (hunk 2 above)
Comment (benoitblanc) [reply docs/docker/readme.md:8]: In original doc, there is a link to `docker/debian` folder, not the `readme.md`. What do you think is the best to link ? Folder or doc about debian docker image ?

Comment (jmkerloch) [reply docs/docker/readme.md:8]: I think it's best to use the doc about debian docker image

Comment (benoitblanc) [reply docs/docker/readme.md]: Updated !

**Review by jmkerloch — APPROVED** [review]
Seems OK to me.

File: requirements/base.txt [hunk 3]
```diff
@@ -1,5 +1,5 @@
 lxml==4.6.4
-psycopg2==2.9.1
+psycopg2-binary==2.9.1
```
Comment (lgrd) [inline requirements/base.txt:2]: Hello, I'm just curious :) why this change ?

File: docs/development/documentation.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+Pour la génération:
+
+```bash
+# install base dependencies
+python -m pip install -U -r requirements/base.txt
+# install aditionnal dependencies
+python -m pip install -U -r requirements/documentation.txt
+# build it
+sphinx-build -b html docs docs/_build
```
Comment (lgrd) [inline docs/development/documentation.md:15]: When I run this command in a container, I have this error 
```
# sphinx-build -b html docs docs/_build
Running Sphinx v5.3.0

Configuration error:
There is a programmable error in your configuration file:

Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/sphinx/config.py", line 350, in eval_config_file
    exec(code, namespace)
  File "/home/docker/docs/conf.py", line 15, in <module>
    from r2gg import __about__
  File "/home/docker/r2gg/__init__.py", line 1, in <module>
    from r2gg._configure import configure
  File "/home/docker/r2gg/_configure.py", line 5, in <module>
    import psycopg2
  File "/usr/local/lib/python3.11/site-packages/psycopg2/__init__.py", line 51, in <module>
    from psycopg2._psycopg import (                     # noqa
SystemError: initialization of _psycopg raised unreported exception
```
I run inside it inside a `docker run -it -v /home/Documents/itineraire/route-graph-generator/:/home/docker/ python:bullseye bash`. It worked for road2 documentation. :/

File: requirements/base.txt (hunk 3 above)
Comment (benoitblanc) [reply requirements/base.txt:2]: It is because `psycopg2-binary` is installed as dependency in Dockerfile here https://github.com/IGNF/route-graph-generator/blob/develop/docker/debian/Dockerfile#L70

Do you know why there are `psycopg2` and `psycopg2-binary` used in container ?

File: docs/development/documentation.md (hunk 4 above)
Comment (benoitblanc) [reply docs/development/documentation.md:15]: You need to install base requirements and documentation requirements before running this command. But there is a problem when I try to install base requirements. I'm looking for the reason...

File: requirements/base.txt (hunk 3 above)
Comment (lgrd) [reply requirements/base.txt:2]: Oh ok ! I understand my misconception. base.txt is for a personn who want to use the project. I forgot that when I asked my question. Indeed, psycopg2 is used for connections to a base inside the r2gg scripts. ^^

File: docs/development/documentation.md (hunk 4 above)
Comment (lgrd) [reply docs/development/documentation.md:15]: I did it (and I had an error with lxml). But this previous error was there. Thanks for looking at it.

Comment (benoitblanc) [reply docs/development/documentation.md:15]: Actually, we need to refactor entry points of r2gg to be able to build documentation without installing base requirements (see #20 )

Comment (lgrd) [reply docs/development/documentation.md:15]: ok !

Comment (benoitblanc) [reply docs/development/documentation.md:15]: @lgrd you should now be able to build documentation without installing base requirements

Comment (lgrd) [reply docs/development/documentation.md:15]: thanks ! it works ;)

**Review by lgrd — APPROVED** [review]
ok for me !
